### PR TITLE
[feature] Add save multiple styles action to style menu

### PR DESCRIPTION
**Pull request by elpaso**
Adds the possibility to save/export multiple styles
at once in QML/SLD and in the DB, in case of file system
export an incremental number is appended to the base
file name i.e. `export.qml`, `export_1.qml` etc.
in case of DBs the user can specify a new name
for the exported syles, and in that case an incremental
number is appended to the base style name i.e. name,
`name 1`, `name 2` etc., if the new name is left blank,
the style name is used and in case of conflicts the
incremental number is appended as well.

ezgif com-optimize

Funded by **ARPA Piemonte**

### CODE REVIEW

File: src/app/qgsvectorlayersavestyledialog.cpp [hunk 1]
```diff
@@ -156,6 +176,55 @@ void QgsVectorLayerSaveStyleDialog::readUiFileContent( const QString &filePath )
   }
 }
 
+void QgsVectorLayerSaveStyleDialog::setupMultipleStyles()
+{
+  // Show/hide part of the UI according to multiple style support
+  if ( ! mSaveOnlyCurrentStyle )
+  {
+    const auto mgr { mLayer->styleManager() };
+    const auto constStyles = mgr->styles();
```
Comment (nyalldawson) [inline src/app/qgsvectorlayersavestyledialog.cpp]: auto :-1:

**Review by nyalldawson — COMMENTED** [review]
Code looks good to me!

File: src/app/qgsvectorlayersavestyledialog.cpp (hunk 1 above)
Comment (nirvn) [reply src/app/qgsvectorlayersavestyledialog.cpp]: Could we try to come up with a coding guideline for auto? As was suggested a few times already, we should limit auto to the for loops iterating through lists / maps really.

Declaring a variable type goes right along using informative/clear variable names.

Comment (elpaso) [reply src/app/qgsvectorlayersavestyledialog.cpp]: I know about the policy: this is the first iteration and I'll clean it up before merge. I was more interested in comments regarding the functionality in this phase. 
But thank you for reminding me.

**Comment by elpaso** [comment]
Merging in its current state, if is there anything I missed please let me know.

**Pull request merged**